x-pack/plugins/observability/server/services/slo/historical_summary_client.test.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClientMock, elasticsearchServiceMock } from '@kbn/core/server/mocks';
+import moment from 'moment';
+import { oneMinute, oneMonth, thirtyDays } from './fixtures/duration';
+import { createSLO } from './fixtures/slo';
+import { DefaultHistoricalSummaryClient } from './historical_summary_client';
+
+const commonEsResponse = {
+  took: 100,
+  timed_out: false,
+  _shards: {
+    total: 0,
+    successful: 0,
+    skipped: 0,
+    failed: 0,
+  },
+  hits: {
+    hits: [],
+  },
+};
+
+const generateEsResponseForRollingSLO = (
+  rollingDays: number = 30,
+  good: number = 97,
+  total: number = 100
+) => {
+  const numberOfBuckets = rollingDays * 2;
+  const day = moment.utc().subtract(numberOfBuckets, 'day').startOf('day');
+  return {
+    ...commonEsResponse,
+    responses: [
+      {
+        ...commonEsResponse,
+        aggregations: {
+          daily: {
+            buckets: Array(numberOfBuckets)
+              .fill(0)
+              .map((_, index) => ({
+                key_as_string: day.clone().add(index, 'day').toISOString(),
+                key: day.clone().add(index, 'day').format('x'),
+                doc_count: 1440,
+                total: {
+                  value: total,
+                },
+                good: {
+                  value: good,
+                },
+                cumulative_good: {
+                  value: good * (index + 1),
+                },
+                cumulative_total: {
+                  value: total * (index + 1),
+                },
+              })),
+          },
+        },
+      },
+    ],
+  };
+};
+
+const generateEsResponseForCalendarAlignedSLO = (good: number = 97, total: number = 100) => {
```

Comment:
💬  Response for calendar aligned SLO starts at the beginning of the SLO current period, up to the current day. For testing purposes, we return only 18 buckets, simulating the 18th of a month.